jcl/src/openj9.dtfj/share/classes/com/ibm/dtfj/image/javacore/JCImage.java

```diff
@@ -235,14 +236,29 @@ public void setcpuType(String cpuType) {
 	}
 
 	/**
-	 * 
+	 * Set the creation time of the image.
+	 * If the time has previously been set using UTC
+	 * then do not modify the time.
 	 * @param creationTime
 	 */
 	public void setCreationTime(long creationTime) {
+		if (!fCreationTimeUTCSet)
+		{
+			fCreationTime = creationTime;
+			fCreationTimeSet = true;
+		}
```

Comment:
This seems inappropriate. There's no way to distinguish whether `getCreationTime()` returns a value in UTC or some other time zone. It may break the interpretation of other times which are relative to the local time captured in the dump.

Reply:
This is about trying to cope when the javacore is parsed in a different time zone to when it was created.

DTFJ API: com.ibm.dtfj.image.Image.getCreationTime()
the image creation time in milliseconds since 1970

So this should and does return a UTC based offset, suitable for constructing a Date(). Trying to define it to be anything else would be confusing.

The existing code takes the javacore time (in the JVM time zone), parses it according to the current (DTFJ process) time zone, converts to UTC, and stores the UTC milliseconds. This is okay unless the time zones differ.

The new code takes the 1TIDATETIMEUTC time if available, parses and stores that as UTC. That seems the right thing to do.
The rest of the new code is an attempt to correct the time where there is no 1TIDATETIMEUTC time.

The flag `fCreationTimeUTCSet` means the time has been set by 1TIDATETIMEUTC and so is accurate and so shouldn't be modified by the fix-up code calling setCreationTime(). There are other ways this could be done, for example adding a method to com.ibm.dtfj.image.javacore.JCImage, but then the fix-up code would need to cast the com.ibm.dtfj.image.Image to a com.ibm.dtfj.image.javacore.JCImage.

Other times in the javacore:
```
1STGCHTYPE     GC History
3STHSTTYPE     14:54:08:695155215 GMT j9mm.101 
```
This is GMT (UTC) and is used for the creation time correction, but is otherwise not used except for a direct return in a trace buffer.

```
1XECTHTYPE     Current thread history (J9VMThread:0x00000000004E4A00)
3XEHSTTYPE     14:54:16:094112611 GMT j9dmp.9 -
```
Similarly, this is GMT (UTC) and is used for the creation time correction, but is otherwise not used except for a direct return in a trace buffer.

This doesn't seem useful:
```
1HKINTERFACE   MM_OMRHookInterface
NULL           ------------------------------------------------------------------------
2HKEVENTID     1
3HKCALLCOUNT     189
3HKTOTALTIME     1205us
3HKLAST          Last Callback
4HKCALLSITE        c:\workspace\openjdk-build\workspace\build\src\openj9\runtime\jcl\common\mgmtinit.c:146
4HKSTARTTIME       Start Time: 1970-01-16T07:17:51.968
```

I hadn't considered this one:
```
1CISTARTTIME   JVM start time: 2023/03/02 at 15:12:54:988
1CISTARTNANO   JVM start nanotime: 718659523066800
```
but that code is unchanged and is parsed in the current time zone into UTC and stored and returned as UTC.
That is okay if the javacore time zone is the same as the time zone of the DTFJ process.